install.sh

```diff
@@ -96,15 +110,30 @@ fi
 
 BASH_IT="$(cd "$(dirname "$0")" && pwd)"
 
-case $OSTYPE in
-  darwin*)
-    CONFIG_FILE=.bash_profile
-    ;;
-  *)
-    CONFIG_FILE=.bashrc
-    ;;
-esac
+# The behaviour set below is that .profile, .bash_login or .bash_profile will be modified to load 
+# .bashrc, and .bashrc will be modified to load bash-it.  This means that .bash-it will work 
+# whether interactive or not (i.e. terminal emulator vs ssh session).
+# This brings the likes of OSX and Solaris into line with the behaviour of various Linux distros
```

Comment:
This is a pretty big show stopper for me. I feel extremely uncomfortable with updating all these files to account for os-specific behavior. Can this be made an interactive decision instead? Preferrably something with os-specific logic.

Reply:
Yes, it's a big change. On one hand, I like the fact that the setup will be more standardized, it might solve some of the issues related to interactive logins...

On the other hand, it's pretty invasive:
* There are more locations in the code where we load/reload OS-specific files, e.g. in the `_bash-it-reload` function (helpers.bash), the `uninstall.sh` file, documentation, etc. - all of these would have to be adjusted as well.
* Updating all of these files would have to be covered in an uninstall scenario as well, we'd have to roll back changes that were created by the Bash-it installer. Not sure how feasible that is.
* Would love to see more (manual) testing on this for other operating systems.